torch/distributed/fsdp/_symbolic_trace.py

```diff
@@ -0,0 +1,220 @@
+            A dict that maps each parameter name to the parameter
+        kind, target, args, kwargs, name, type_expr: inputs to the ``create_proxy`` function.
+    """
+    proxy = create_proxy(kind, target, args, kwargs, name, type_expr)
+
+    if kind in ["call_function", "call_method"]:
+        if args is not None:
+            named_params: List[Tuple[str, torch.nn.Parameter]] = []
+            for arg in args:
+                if isinstance(arg, torch.fx.Proxy) and arg.node.target in params_dict:
+                    param = params_dict[arg.node.target]
+                    named_params.append((arg.node.target, param))
+                    if param not in execution_info._traced_param_set:
+                        execution_info.param_exec_order.append(param)
+                        execution_info._traced_param_set.add(param)
+            if named_params:
+                execution_info.module_execution_info_dict[
+                    execution_info.current_module
+                ].append((execution_info.current_module, named_params))
+    elif kind == "call_module":
+        module = execution_info.current_module
+        named_params_list = list(module.named_parameters())
+        if named_params_list != []:
+            execution_info.module_execution_info_dict[module].append(
+                (module, named_params_list)
+            )
+        for (_, p) in named_params_list:
+            if p not in execution_info._traced_param_set:
+                execution_info.param_exec_order.append(p)
+                execution_info._traced_param_set.add(p)
+    return proxy
+
+
+def _patched_call_module(
+    call_module: Callable,
+    execution_info: _ExecutionInfo,
+    module: torch.nn.Module,
+    forward: Callable[..., Any],
+    args: Tuple[Any, ...],
+    kwargs: Dict[str, Any],
+) -> Any:
+    """
+    Override of :meth:`~torch.fx.Tracer.call_module`.
+    ``Tracer.call_module`` is called in symbolic tracing for each non-root module.
+    This override intercepts the recording of each operation and
+    update ``execution_info.module_forward_order`` and ``execution_info.module_execution_info_dict``.
+
+    Args:
+        call_module (Callable):
+            The ``call_module`` function to be patched.
+        execution_info (_ExecutionInfo):
+            Used to repord the execution information.
+        module, forward, args, kwargs: inputs to the ``call_module`` function.
+    """
+    execution_info.module_forward_order.append(module)
+    named_params_list = list(module.named_parameters())
+    if named_params_list != []:
+        execution_info.module_execution_info_dict[execution_info.current_module].append(
+            (module, list(module.named_parameters()))
+        )
+    # Stores away current_module for restoration later
+    old_current_module = execution_info.current_module
+    execution_info.current_module = module
+    # Initialize execution_info.module_execution_info_dict[module]. Note that if
```

Comment:
nit: no need for comment
```suggestion
    # Note that if
```